src/components/export-execution/resource-file-preview.tsx

```diff
@@ -0,0 +1,39 @@
+import { Center, Pagination, ScrollArea, Title, Mark } from '@mantine/core';
+import { CodeHighlight } from '@mantine/code-highlight';
+import { useState } from 'react';
+import { ResourceFileInfo } from './requested-files';
+import { stringifyJsonArray as stringifyJsonObjectArray } from '@/util/ndjsonParsers';
+
+import classes from '@/app/global.module.css';
+
+export interface ResourceFilePreviewProps {
+  file?: ResourceFileInfo;
+}
+
+// Component for preview modal that renders the json code for each resource in a file.
+export default function ResourceFilePreview({ file }: ResourceFilePreviewProps) {
+  if (!file) return <></>;
```

Comment:
Will this ever happen? If so, maybe handle this error where this modal is called and that way file is not optional

Reply:
Good call again